src/components/home/index.js

```diff
@@ -16,9 +21,35 @@ import {
 
 const Home = () => {
   const { showMemberRoleUpdateModal, isSuperUserMode } = userContext();
+  const {
+    state: { newMembers: newMembersList },
+  } = membersContext();
+  const { searchTerm } = searchMemberContext();
+  const filterMembersList = searchMembers(newMembersList, searchTerm);
+
   const { query } = useRouter() || { query: { dev: false } };
   const { dev } = query;
   const [isOptionKey, setIsOptionKey] = useState(false);
+  const [newMembers, setNewMembers] = useState([]);
+  const [scrollCount, setScrollCount] = useState(0);
+
+  const membersToShow = 10;
+
+  const fetchNewMembers = () => {
+    const newMemberArr = searchTerm ? filterMembersList : newMembersList;
+    const noOfMembersToShow = scrollCount * membersToShow;
+    const slicedNewMembersList = newMemberArr.slice(0, noOfMembersToShow);
+    setNewMembers(slicedNewMembersList);
+    setScrollCount(scrollCount + 1);
+  };
+
+  useEffect(() => {
+    fetchNewMembers();
+  }, [newMembersList]);
+
+  const loadMoreData = () => {
+    fetchNewMembers();
+  };
```

Comment:
Do we really need this?
On line 81, can we simply say
 `{newMembers && <Waypoint onEnter={fetchNewMembers} />}`

Reply:
Makes sense.